apps/fishing-map/features/map/map.hooks.ts

```diff
@@ -78,21 +78,18 @@ export const useGeneratorsConnect = () => {
   const showTimeComparison = useSelector(selectShowTimeComparison)
   const timeComparisonValues = useSelector(selectTimeComparisonValues)
 
-  // TODO memoize allIds?
-  const allIds = generatorsConfig.map((g) => g.id)
-  const layersLoaded = useMapSourceTilesLoaded(allIds || [], true)
-  const layersLoadedSerialized = (layersLoaded as boolean[]).join('')
-  const updatedGeneratosConfig = useMemo(() => {
+  const sourceTilesLoaded = useMapSourceTiles()
+  const updatedGeneratorConfig = useMemo(() => {
     return generatorsConfig.map((generatorConfig, i) => {
       if (generatorConfig.type === GeneratorType.Polygons) {
         return {
           ...generatorConfig,
-          loaded: layersLoaded[i],
+          opacity: sourceTilesLoaded[generatorConfig.id]?.loaded ? 1 : 0,
         }
       }
       return generatorConfig
     })
-  }, [generatorsConfig, layersLoadedSerialized])
+  }, [generatorsConfig])
```

Comment:
Are you sure you don't need to add `sourceTilesLoaded` in the dependency array?

Reply:
This is what the linter says... and I've tested and it works as expected 🤷‍♂️
![image](https://user-images.githubusercontent.com/10500650/169293905-0ddeba62-21e5-4c92-8717-43eb0ecf2a04.png)